chore: update xud-docker link to docs

kilrau opened the pull request:
Update link from 
https://docs.exchangeunion.com/start-trading/user-guide
to
https://docs.exchangeunion.com/start-earning/market-maker-guide#the-setup
since market makers are the target user group of xud-docker

raladev commented:
I disagree with #the-setup anchor. it moves me to light-setup section vs full setup section, but I know nothing about that because the explanation a little bit above.

Solutions:
1. Delete anchor;
2. Move Two Modes section into Setup section (more preferable).

Wdyt @kilrau  ?

kilrau commented:
The Light/Full Section will disappear after https://github.com/ExchangeUnion/xud-docker/pull/366 is in, but ok I think linking to https://docs.exchangeunion.com/start-earning/market-maker-guide is cleaner in general, so I fixed that. I also adjusted the readme wording a bit to make it clear, that xud-docker is a product for market makers.
